lib/metadataTypes/MobileKeyword.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const MetadataType = require('./MetadataType');
+const Util = require('../util/util');
+
+/**
+ * MobileKeyword MetadataType
+ * @augments MetadataType
+ */
+class MobileKeyword extends MetadataType {
+    /**
+     * Retrieves Metadata of Mobile Keywords
+     * Endpoint /legacy/v1/beta/mobile/keyword/ return all Mobile Keywords with all details.
+     * @param {String} retrieveDir Directory where retrieved metadata directory will be saved
+     * @returns {Promise<Object>} Promise of metadata
+     */
+    static retrieve(retrieveDir) {
+        return super.retrieveREST(retrieveDir, '/legacy/v1/beta/mobile/keyword/?view=simple', null);
+    }
+
+    /**
+     * Retrieves event definition metadata for caching
+     * @returns {Promise<Object>} Promise of metadata
+     */
+    static retrieveForCache() {
+        return super.retrieveREST(null, '/legacy/v1/beta/mobile/keyword/?view=simple', null);
+    }
+
+    /**
+     * Retrieve a specific keyword
+     * @param {String} templateDir Directory where retrieved metadata directory will be saved
+     * @param {String} name name of the metadata file
+     * @param {Object} templateVariables variables to be replaced in the metadata
+     * @returns {Promise<Object>} Promise of metadata
+     */
+    static async retrieveAsTemplate(templateDir, name, templateVariables) {
+        return super.retrieveREST(
+            templateDir,
+            `/legacy/v1/beta/mobile/keyword/?view=simple&$where=keyword%20eq%20%27${name}%27%20`,
+            null,
+            templateVariables
+        );
+        // TODO: Decide if we want to keep default handling (and move other types) or keep unique parsing?
```

Comment:
lets have a separate clean-up task for consolidating the typed rt into one generic one and keep this one here - I'm assuming you would then uncomment the following lines?

Reply:
so i tried something a bit half way

Reply:
lets have a call for this... tried understanding what you wrote there but failed